Infrastructure/Persistence/EfCoreRepository/UserRepository.cs

```diff
@@ -21,9 +23,10 @@ public bool IsExitByEmail(string email)
             throw new NotImplementedException();
         }
 
-        public Task<User> RegisterUserAsync(User user)
+        public async Task<User> RegisterUserAsync(User user)
         {
-            throw new NotImplementedException();
+           await _applicationContext.AddAsync(user);
```

Comment:
I noticed that the **ApplicationContext** class need to undergo some updates. which am working on currently, Given these changes, we need to restructure this code to ensure it remains compatible and maintainable.